compiler/rustc_expand/src/mbe/metavar_expr.rs

```diff
@@ -41,6 +45,16 @@ impl MetaVarExpr {
         check_trailing_token(&mut tts, sess)?;
         let mut iter = args.trees();
         let rslt = match ident.as_str() {
+            "concat" => {
+                let lhs_is_var = try_eat_dollar(&mut iter);
+                let lhs_ident = parse_ident(&mut iter, sess, ident.span)?;
+                if !try_eat_comma(&mut iter) {
```

Comment:
Do we even need commas? Referencing `paste` again, the syntax is 

```rust
[<Q R S T>] // turns into QRST
```

Which is somewhat easier to read without the commas. (Similar to C `Q ## R ## S ## T`, though that definitely isn't easier to read)

Reply:
The comma was added to follow the syntax of the other meta-variable expressions. For example: `count($foo, 0)`.

In my personal opinion, the matter about case conversion and omission of commas should be discussed at a later opportunity.

Reply:
@c410-f3r 
Could you list this concern on the tracking issue?
(And maybe create a new tracking issue instead of https://github.com/rust-lang/rust/issues/29599.)